packages/flutter_tools/lib/src/cache.dart

```diff
@@ -535,12 +538,19 @@ abstract class EngineCachedArtifact extends CachedArtifact {
 
       _makeFilesExecutable(dir);
 
+      // TODO(jonahwilliams): clean this logic up to better handle different framework names.
       final File frameworkZip = fs.file(fs.path.join(dir.path, 'Flutter.framework.zip'));
       if (frameworkZip.existsSync()) {
         final Directory framework = fs.directory(fs.path.join(dir.path, 'Flutter.framework'));
         framework.createSync();
         os.unzip(frameworkZip, framework);
       }
+      final File desktopFrameworkZip = fs.file(fs.path.join(dir.path, 'FlutterMacOS.framework.zip'));
```

Comment:
Since I'm not familiar with the overall structure of this code: why does the macOS framework need to be specified/handled in two different places? It's both here and in the artifact list later. The iOS code doesn't seem to have the same structure (it's not obvious to me why Flutter.framework has special handling here, but it's not in the list below like the macOS version is.)

Reply:
Its a bit of a historical artifact (har-har), I can't quite remove it yet but eventually this will be gone. TLDR: I am scared to touch it